src/api/v4/error.js

```diff
@@ -13,13 +13,23 @@ function CartoError (error) {
     this.message = error.message;
     return this;
   }
+  if (error && error.responseText) {
+    this.message = _handleAjaxResponse(error);
+    return this;
+  }
+
   this.message = 'unexpected error';
 }
 
 function _isWindshaftError (error) {
   return error && error.errors_with_context;
 }
 
+function _handleAjaxResponse (error) {
+  var dataviewError = JSON.parse(error.responseText);
```

Comment:
I would make this a bit more generic (not mentioning dataviews). Also, we are assuming `error.responseText` is a JSON. Perhaps we can add a try...catch around `JSON.parse` and make sure `response.errors` is there and is an array, otherwise return a generic error.